aiida/cmdline/commands/cmd_status.py

```diff
@@ -74,43 +74,31 @@ def verdi_status():
     except Exception as exc:
         print_status(ServiceStatus.ERROR, 'repository', "Error with repo folder", exception=exc)
         exit_code = ExitCode.CRITICAL
+    else:
+        print_status(ServiceStatus.UP, 'repository', repo_folder)
 
-    print_status(ServiceStatus.UP, 'repository', repo_folder)
-
-    # getting the postgres status
+    # Getting the postgres status by trying to get a database cursor
     try:
-        with Capturing(capture_stderr=True):
-
-            postgres = Postgres.from_profile(profile)
-
-        dbinfo = postgres.dbinfo.copy()
-        if postgres.is_connected:
-            print_status(ServiceStatus.UP, 'postgres', "Connected to {}@{}:{}".format(
-                dbinfo['user'], dbinfo['host'], dbinfo['port']))
-        else:
-            print_status(ServiceStatus.DOWN, 'postgres', "Unable to connect to {}:{}".format(
-                dbinfo['host'], dbinfo['port']))
-            exit_code = ExitCode.CRITICAL
-
-    except Exception as exc:
-        print_status(
-            ServiceStatus.ERROR,
-            'postgres',
-            "Error connecting to {}:{}".format(profile.database_hostname, profile.database_port),
-            exception=exc)
+        backend = manager.get_backend()
+        backend.cursor()
+    except Exception:
+        database_data = [profile.database_hostname, profile.database_port]
+        print_status(ServiceStatus.DOWN, 'postgres', "Unable to connect to {}:{}".format(*database_data))
```

Comment:
since this is now (correctly) using all the info from the profile, the error message can also include the username (like the success message below).
this means you can move `database_data` outside and reuse it in both places.